crates/wasmprinter/src/operator.rs

```diff
@@ -316,6 +316,30 @@ macro_rules! define_visit {
             )?;
         }
     );
+    (payload $self:ident RefTestNonNull $hty:ident) => (
+        $self.push_str(" ");
+        let rty = RefType::new(false, $hty)
+            .ok_or_else(|| anyhow!("implementation limit: type index too large"))?;
+        $self.printer.print_reftype(rty)?;
+    );
+    (payload $self:ident RefTestNullable $hty:ident) => (
+        $self.push_str(" ");
+        let rty = RefType::new(true, $hty)
+            .ok_or_else(|| anyhow!("implementation limit: type index too large"))?;
+        $self.printer.print_reftype(rty)?;
+    );
+    (payload $self:ident RefCastNonNull $hty:ident) => (
+        $self.push_str(" ");
+        let rty = RefType::new(false, $hty)
+            .ok_or_else(|| anyhow!("implementation limit: type index too large"))?;
+        $self.printer.print_reftype(rty)?;
+    );
+    (payload $self:ident RefCastNullable $hty:ident) => (
+        $self.push_str(" ");
+        let rty = RefType::new(true, $hty)
+            .ok_or_else(|| anyhow!("implementation limit: type index too large"))?;
+        $self.printer.print_reftype(rty)?;
+    );
```

Comment:
Huh. Interesting! I think this is the first instance of two different opcodes having the same name. That seems not great from an AST/tooling perspective... you wouldn't happen to know if this is being bikeshedded and/or discussed elsewhere would you?

Reply:
I wasn't aware of any discussion and didn't find anything when searching. It is definitely a bit funky.

Reply:
Filed https://github.com/WebAssembly/gc/issues/491 to ask for some background